### PR TITLE
feat: add page transitions

### PR DESCRIPTION
**Pull request by fsdiogo**
This PR supersedes https://github.com/moxystudio/rfcs-oss/pull/4.

### CODE REVIEW

**Comment by satazor** [comment]
@fsdiogo one of the drawbacks of highway was that it doesn’t handle scroll restoration integrated with the animations. It seems that this RFC doesn’t state how we solve that. Did you find or think of solutions to tackle that?

**Comment by fsdiogo** [comment]
@satazor I didn't mention the scroll restoration in this RFC because it's a non-issue as Framer Motion takes care of it.

You can try it [here](https://next-with-moxy.fsdiogo.now.sh/).

**Comment by satazor** [comment]
@rafaelramalho19 Your input would be valuable to this RFC. Can you quickly evaluate the proposed library in comparison to others you have used in the past? The library to use is important as it will be probably be the same for animations outside page transitions, to keep the codebase small.

@fsdiogo feel free to tag the RFC as final-comments if you think it’s ready on your part.

File: rfcs/0006-page-transitions.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+- Add drag, pan, hover and tap gestures.
+- Respond to gestures with animations.
+- Deeply animate throughout React trees via variants.
+
+A good place to put these would be in the [`www/shared/components`](https://github.com/moxystudio/next-with-moxy/tree/master/www/shared/components), in a *transitions* folder for example.
+
+```js
+import React from 'react';
+import PropTypes from 'prop-types';
+import { motion } from 'framer-motion';
```
Comment (rafaelramalho19) [inline rfcs/0006-page-transitions.md:63]: Is it really worth to include such a big lib just for page transitions?

Motion has a lot of code just for Drag and Spring physics, but do we really need them when transitioning between pages?

A simple css only solution would be better for the use-case represented in this example:

```css
@keyframes pageEnter {
    from {
         opacity: 0;
         transform: translateX(-600px)
    }
    to {
        opacity: 1; transform: translateX(0);
    }
}

@keyframes pageExit {
    from {
         opacity: 1;
         transform: translateY(0)
    }
    to {
        opacity: 0; transform: translateY(300px);
    }
}

.page.enter {
    animation: pageEnter 0.8s ease both;
}

.page.exit {
    animation: pageExit 0.5s ease forwards;
}
```


I know that you can argue that "it's simpler and faster to use js only with a custom lib". But is it really worth the extra bundle size?

Comment (rafaelramalho19) [reply rfcs/0006-page-transitions.md:63]: More data into my explanation:
https://bundlephobia.com/result?p=framer-motion@1.8.3

Comment (satazor) [reply rfcs/0006-page-transitions.md:63]: I haven’t analyzed it, but perhaps the package is modular and we can import only what we need?

**Comment by fsdiogo** [comment]
Untagged the `final-comments` as we're going to make a list of requirements and test libs to check if they match those requirements.

**Comment by rafaelramalho19** [comment]
@satazor I think you can close this?

**Comment by satazor** [comment]
Yep we can. There's https://github.com/moxystudio/react-page-swapper now!

**Pull request closed without merging**